### ISSUE TITLE
gpxinfo AttributeError: GPX instance has no attribute 'author'

### ISSUE THREAD

**Issue by szo1215** [tkrajina/gpxpy#48]
Hi. I'm using gpxpy very well. but When I analyzed my strava gpxfile, It didn't work because there was no author information in this file. In this case What should i do?

**Comment by tkrajina** [comment]
Can you send me an example track to test this?

**Comment by szo1215** [comment]
It is strava gpx file.
Thanks.

2015-10-27 2:52 GMT+09:00 Tomo Krajina <notifications@github.com>:

**Comment by tkrajina** [comment]
I tried to parse and convert back to XML, and to me it looks like it works:

    In [1]: import gpxpy

    In [4]: gpx = gpxpy.parse(file('test_files/gpx1.1_with_all_fields.gpx'))

    In [5]: gpx.author_name
    Out[5]: 'author name'

    In [6]: gpx = gpxpy.parse(file('test_files/gpx1.0_with_all_fields.gpx'))

    In [7]: gpx.author_name
    Out[7]: 'example author'

> It is strava gpx file.

I know it's a Strava file, you already said that. I asked if you can send me that file. I don't plan to experiment with Strava and figure out what "It didn't work" means.  If you already have a file -- why wouldn't you help me by sending it? Also, a couple of lines with your gpxpy code is welcome.

I know this may sound rude, but keep in mind that I *am* willing to help/fix your bug. "It's a strava file" and "It didn't work" just isn't a valid bug request.

**Comment by szo1215** [comment]
I don't have experience about sending open source issue. so I did that.
If I were send other bug request, I would get more cautious.
I'm sorry.

**Comment by jwm-art-net** [comment]
Have same issue with GPX files which have been converted from TCX files by gpsbabel. The TCX files were converted from Garmin FIT files by fittotcx.
http://pastebin.com/NhXC6xVX

**Comment by tkrajina** [comment]
Note that the gpx1.1 implementation has no *simple* `<author>` property. That's why the new gpxpy (1.0+) has a slightly different object model. The properties in the gpxpy are made to work with both gpx1.0 and gpx1.1. That's why the author *fields* are:
```
        self.author_name = None
        self.author_email = None
        self.author_link = None
        self.author_link_text = None
        self.author_link_type = None
```
If you try to add `<author>qwe</author>` in your file and check `gpx.author_name` (*not* `gpx.author`) it should be `qwe`.

**Comment by jwm-art-net** [comment]
I'm not using the Library API but the gpxinfo command line tool. I don't really follow what you're saying. I added `<author>qwe</author>` to the file and re-ran gpxinfo and still get the above error message. Then I converted a TCX into a GPX 1.1 thinking that would help but still the same 'no author' error.

It seems to me that the author attribute is optional and the case where it is missing should be handled gracefully. Is that not correct?
